cpp/src/io/orc/writer_impl.cu

```diff
@@ -38,6 +38,13 @@ namespace orc {
 using namespace cudf::io::orc;
 using namespace cudf::io;
 
+struct row_group_index_info {
+  int32_t pos;        // Position
+  int32_t blk_pos;    // Block Position
+  int32_t comp_pos;   // Compressed Position
+  int32_t comp_size;  // Compressed size
```

Comment:
Feels like these should be uin32_t + a bool to indicate if the object is valid

Reply:
If it is uint32_t, we will not be able to store have -1 as default value

Reply:
And there are chances were one of the values are set and other is not, in that case it would be hard to discern.

Reply:
yes, but then we don't need the default value, we can use a separate bool to indicate validity (unless I'm missing the meaning of -1).

Reply:
my bad, looks like a record has two "layers" of validity. Darn.